### PR TITLE
Added check for overwriting previous runs

### PR DESCRIPTION
**Pull request by evanmlim**
Added option to enable validation loss graph. Graph data is also saved to file (format can be changed). Added assert statement at start of main.py to check if the specified results folder contains a previous run to prevent accidentally overwriting it.

### CODE REVIEW

**Comment by aecelaya** [comment]
The training and validation loss curves are already recorded using Tensorboard in the ./results/logs folder. Are you not able to access them? What is the purpose of the validation graph option?

For the assert statement for the results.csv file, I think that it would make more sense to append the date/time stamp to the results.csv file so that it's something like results_07-16-2024-2328. Otherwise, I think that this assertion would cause more problems than it would solve. For example, what happens if you do a test run and then want to go back and run the same experiment for longer? It seems a little cumbersome to have users go back and delete the results.csv file every time.

**Comment by evanmlim** [comment]
Oh. I didn't know that the loss curves were stored somewhere. I'll start using that. (I'm going to be honest, I somehow never touched the logs folder)

For the assert statement, I've had many cases where I forget to change the results location and I end up overwriting a previous run which had data I still needed. I don't think just renaming the results csv file will work because there is other important data that is also save-worthy (Like if you want to compare predictions from an old run to a newer run). Maybe instead prompt the user that there is a previous run like: "There is a previous run in the results folder. Overwrite? (y/n)". We can also add an option to disable this prompt.

**Comment by aecelaya** [comment]
Hmm how about we add a flag like --overwrite and set the default value to False? In that case, it throws the assertion error you implemented. Otherwise it overwrites the results. What do you think?

**Comment by evanmlim** [comment]
That sounds good! I'll add the flag and revert the validation graph changes.

**Review by aecelaya — APPROVED** [review]

**Comment by aecelaya** [comment]
Done! There are a few more changes that I'm working on right now (not related to this PR) that I want to add soon. Once those are committed, I'll release a new version with this change.
